torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -519,13 +543,58 @@ void XLAGraphExecutor::TensorCollectionBarrier(SyncTensorCollection* coll) {
 
 std::vector<torch::lazy::BackendDataPtr>
 XLAGraphExecutor::ExecuteComputationWithBarrier(
-    torch::lazy::ComputationPtr computation,
-    c10::ArrayRef<torch::lazy::BackendDataPtr> arguments,
+    torch::lazy::hash_t hash,
+    std::vector<torch::lazy::BackendDataPtr> arguments,
```

Comment:
Should this be `const std::vector<torch::lazy::BackendDataPtr>&`? Otherwise, a copy will be introduced.

Reply:
I actually do this intentionally, if you check the code below, `arguments` is `std::move` to the `async` object because we want to maintain a copy for the execution thread.

Reply:
Sounds good.